### PR TITLE
Infer dims & coords from xarray variables passed to pm.Data

### PR DESCRIPTION
**Pull request by mjhajharia**
addresses #5791 

- [x] Inferring dims from named pandas indexes
- [x] Inferring dims from DataArrays
- [x] Inferring coords from DataArrays
- [ ] A unit test
- [ ] Possibly a (backwards compatible with deprecation warning) rename of the export_index_as_coords to infer_dims_and_coords.

cc: @michaelosthege (I might have made errors, also I just emulated the pandas code for xarray, not sure if it makes sense)

### CODE REVIEW

File: pymc/data.py [hunk 1]
```diff
@@ -485,6 +485,8 @@ def determine_coords(
             dim_name = value.index.name
         if dim_name is not None:
             coords[dim_name] = value.index
+        if dims is None: 
+            dims = dims_name
```
Comment (michaelosthege) [inline pymc/data.py:489]: `dims` should be a list

In addition, it might be a good idea to account for the case of Multi-Indexed DataFrames. (`raise`?)

**Review by michaelosthege — COMMENTED** [review]
sorry @mjhajharia, I forgot to submit the review commend I made a few days ago 🙈

It looks like there's a block of code that got duplicated?

**Comment by ricardoV94** [comment]
Is this PR salvageable or should we close it?

**Comment by michaelosthege** [comment]
Actually this has been done in https://github.com/pymc-devs/pymc/pull/6514, so we can close

**Pull request closed without merging**